packages/fields-mongoid/src/Implementation.js

```diff
@@ -92,15 +92,22 @@ export class KnexMongoIdInterface extends KnexFieldAdapter {
 
   setupHooks({ addPreSaveHook, addPostReadHook }) {
     addPreSaveHook(item => {
-      const valType = typeof item[this.path];
+      // Only run the hook if the item actually contains the field
+      // NOTE: Can't use hasOwnProperty here, as the mongoose data object
+      // returned isn't a POJO
+      if (!(this.path in item)) {
+        return item;
+      }
 
-      if (item[this.path] && valType === 'string') {
-        item[this.path] = normaliseValue(item[this.path]);
-      } else if (!item[this.path] || valType === 'undefined') {
-        delete item[this.path];
+      if (item[this.path]) {
+        if (typeof item[this.path] === 'string') {
+          item[this.path] = normaliseValue(item[this.path]);
+        } else {
+          // Should have been caught by the validator??
+          throw `Invalid MongoID value given for '${this.path}'`;
+        }
       } else {
-        // Should have been caught by the validator??
-        throw `Invalid value given for '${this.path}'`;
+        item[this.path] = null;
```

Comment:
After discovering the bug in the `Decimal` field, I went back and rewrote the other fields which do the same logic to make it a bit clearer (to my brain). This should be functionally identical to what it was before.